Modules/_functoolsmodule.c

```diff
@@ -751,7 +751,7 @@ lru_cache_make_key(PyObject *args, PyObject *kwds, int typed)
     Py_ssize_t key_size, pos, key_pos, kwds_size;
 
     /* short path, key will match args anyway, which is a tuple */
-    if (!typed && !kwds) {
+    if (!typed && (!kwds || PyDict_GET_SIZE(kwds) == 0)) {
```

Comment:
You can move the definition of `kwds_size` up and use it here.
```
if (!typed && !kwds_size) {
```

Reply:
Good suggestion.  Done.